catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/ReindexCommand.java

```diff
@@ -0,0 +1,492 @@
+      if (!publishExecutor.isShutdown()) {
+        if (force) {
+          publishExecutor.shutdownNow();
+        } else {
+          publishExecutor.shutdown();
+        }
+      }
+    }
+  }
+
+  private void waitForCompletion() throws InterruptedException {
+    boolean running = true;
+    if (readerThread == null) {
+      return;
+    }
+
+    while (running) {
+      if (!readerThread.isRunning()) {
+        publishExecutor.shutdown();
+        publishExecutor.awaitTermination(15, TimeUnit.MINUTES);
+        running = false;
+      }
+      Thread.sleep(1000);
+    }
+  }
+
+  private List<Metacard> getData(org.codice.solr.client.solrj.SolrClient sourceSolr)
+      throws IOException, SolrServerException {
+    final SolrQuery query = getQuery();
+    List<Metacard> data = new ArrayList<>();
+
+    LOGGER.trace("Retrieving data with query: {}", query);
+
+    /**
+     * Always query the default collection, unable to use overload function that takes in a core
+     * name due to how HttpSolrClientFactory create its client
+     * https://github.com/codice/ddf/blob/master/platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java#L130
+     * this limitation also spill over when using solr Cloud
+     */
+    QueryResponse response = sourceSolr.query(query);
+    cursorMark = response.getNextCursorMark();
+    SolrDocumentList docList = response.getResults();
+
+    if (LOGGER.isTraceEnabled()) {
+      LOGGER.trace("Query returned: {} items", docList.size());
+    }
+
+    for (SolrDocument doc : docList) {
+      try {
+        Metacard metacard = metacardClient.createMetacard(doc);
+        data.add(metacard);
+      } catch (MetacardCreationException e) {
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.debug("Unable to convert: {} to metacard", doc, e);
+        }
+      }
+    }
+    return data;
+  }
+
+  @VisibleForTesting
+  SolrQuery getQuery() {
+    StringBuilder querySB = new StringBuilder();
+    if (StringUtils.isNotBlank(afterDate) && StringUtils.isNotBlank(beforeDate)) {
+      querySB.append("[").append(afterDate).append(" TO ").append(beforeDate).append("]");
+    } else if (StringUtils.isNotBlank(afterDate) && StringUtils.isBlank(beforeDate)) {
+      querySB.append("[").append(afterDate).append(" TO NOW").append("]");
+    } else if (StringUtils.isBlank(afterDate) && StringUtils.isNotBlank(beforeDate)) {
+      querySB.append("[").append(EARLY_TIME).append(" TO ").append(beforeDate).append("]");
+    }
+
+    SolrQuery query;
+    if (querySB.length() > 0) {
+      query = new SolrQuery(field + ":" + querySB.toString());
+    } else {
+      query = new SolrQuery("*:*");
+    }
+
+    SortClause sort = new SortClause("metacard.created_tdt", ORDER.desc);
+    query.setSort(sort);
+    query.addSort(new SortClause("id_txt", ORDER.desc));
+    query.setParam(CursorMarkParams.CURSOR_MARK_PARAM, cursorMark);
+    query.setRows(Math.max(PAGE_SIZE, WRITE_TXN_SIZE));
```

Comment:
:question: Won't this always be `PAGE_SIZE` since `WRITE_TXN_SIZE` is always smaller than `PAGE_SIZE`?  Is this here just in case someone updates the constants?